lib/routes/account.js

```diff
@@ -877,26 +879,26 @@ module.exports = (
             && ! doSigninConfirmation
             && emailRecord.emailVerified
           if (shouldSendNewDeviceLoginEmail) {
-            return getGeoData(ip)
-              .then(
-                function (geoData) {
-                  mailer.sendNewDeviceLoginNotification(
-                    [],
-                    emailRecord,
-                    {
-                      acceptLanguage: request.app.acceptLanguage,
-                      flowId: flowId,
-                      flowBeginTime: flowBeginTime,
-                      ip: ip,
-                      location: geoData.location,
-                      timeZone: geoData.timeZone,
-                      uaBrowser: sessionToken.uaBrowser,
-                      uaBrowserVersion: sessionToken.uaBrowserVersion,
-                      uaOS: sessionToken.uaOS,
-                      uaOSVersion: sessionToken.uaOSVersion,
-                      uaDeviceType: sessionToken.uaDeviceType
-                    }
-                  )
+            const promises = [getGeoData(ip), features.isSecondaryEmailEnabled() ? db.accountEmails(sessionToken.uid) : P.resolve([])]
```

Comment:
I think you should be able to do `return P.all([getGeoData(ip), secondaryEmails])` where `secondaryEmails` is a promise set above, this way you don't have to worry about naming `promises`